controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -88,64 +95,67 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
         return Controller.NodeUri.newBuilder().setEndpoint(host.getIpAddr()).setPort(host.getPort()).build();
     }
 
+    private void closeConnection(Reply reply, RawClient client) {
+        log.info("Closing connection as a result of receiving: {}", reply);
+        RawClient c;
+        synchronized (lock) {
+            c = client;
+            client = null;
+        }
+        if (c != null) {
```

Comment:
The only way c could be null here is if the client passed into this method was null in the first place.